add-chain/flags/flags.go

```diff
@@ -70,6 +70,18 @@ var (
 		Usage:    "Filepath to rollup.json input file",
 		Required: true,
 	}
+	DeployConfigFlag = &cli.StringFlag{
+		Name:     "deploy-config",
+		EnvVars:  prefixEnvVars("DEPLOY_CONFIG"),
+		Usage:    "Filepath to deploy-config json input file",
+		Required: true,
+	}
+	GenesisCreationCommit = &cli.StringFlag{
+		Name:     "genesis-creation-commit",
+		EnvVars:  prefixEnvVars("GENESIS_CREATION_COMMIT"),
+		Usage:    "Commit in the https://github.com/ethereum-optimism/optimism/ repo at which the chain's genesis was created",
+		Required: true,
+	}
```

Comment:
It would be a much better user experience, and less error-prone, if this commit was written into the deploy-config or rollup.json file. If we put the onus on the user to remember which commit they used, we are much more likely to have issues than if we automatically record that information via the tooling the user already needs to use (i.e. `op-node genesis l2` command)

Reply:
Agreed that we should have the genesis creation tool output the commit. I'm not sure whether one of the existing files or a new one would be best.

A real small lift would be to print a message: "the commit you used just now might be important for future validation --- remember it!". Not an ideal solution, but maybe not a crazy stop gap until we get something better in place?

There are also plans afoot to approach this in a way which doesn't require such complex validation https://github.com/ethereum-optimism/specs/issues/350. So if that happens, the need to remember the commit would (probably?) go away.

Reply:
Not sure if I have suggested this in one of the discussions - the genesis creation tool could create a git tag with a "well known" format for the commit ex: `chainName-networkName-deploy`. Once that is done, it removes the need to remember the commit and any tools that run in the future could rely on the well known name to determine the previous point in time code.

Reply:
@vdamle that is an interesting suggestion, I suppose one downside is that the monorepo needs to permit a large number of contributors to push git tags, and the list of tags will get pretty long. I think the plan of asking operators to launch off one of a small list of "released" commits would be a better solution, and also reduces the work it will take to validate the chain in future.

Reply:
Whatever approach we take in the future, we are stuck with the legacy behaviour now for quite a few chains -- so I don't think there is much to do on this PR to address the problem.